tests/test_calibration.py

```diff
@@ -57,6 +57,10 @@ def test_AGIPD_CalibrationData_metadata():
     assert isinstance(agipd_cd["Offset", "AGIPD00"], SingleConstant)
     assert agipd_cd["Offset", "Q1M2"] == agipd_cd["Offset", "AGIPD01"]
 
+    bva = agipd_cd["Offset", "AGIPD00"].metadata("begin_validity_at")
```

Comment:
It seems to me these tests do not check the "no-CCV-ID" case for both `.metadata()` and `.metadata_dict()`. Is this state _currently_ possible, or would it only happen for potential future cases (you mentioned `calibration_metadata.yml`)?

Reply:
It's basically a potential future scenario - e.g. if you've just created a new batch of constants in a folder, and you want to make a `CalibrationData` object to look at them before injecting them to CalCat. So my aim is that the design won't need breaking changes if/when we do that, rather than making that work now.